### ISSUE TITLE
Sync does not work with crash-looping container anymore (since 4.13)

### ISSUE THREAD

**Issue by BirknerAlex** [devspace-cloud/devspace#1124]
**What happened?**  

I upgraded from 4.12 to 4.13 today. Running `devspace dev` does sync my golang application to the container. Since the new version 4.13 the sync breaks if the container is in a crash-loop due a syntax error.

In 4.12 I was able to fix the code issue and `devspace dev` synced  the change (magically?) to the crash-looping container and restarted it with the fixed version. Now I need to re-run the `devspace dev` command in such case and wait since devspace can't sync the fixed code inside the crashing container?

Logs: 

```
[sc1-backend-api] 
[sc1-backend-api] ############### Restart container ###############
[sc1-backend-api] 
[sc1-backend-api] # syscore.one/backend/api/pkg/api
[sc1-backend-api] api/pkg/api/project.go:226:1: syntax error: unexpected EOF, expecting }
[error]  Error restarting sync: Error selecting pod: Pod 'sc1-backend-api-65c45967f4-d5qbv' has critical status 'Error' 

To get more information about this pod's status, run: kubectl describe pod sc1-backend-api-65c45967f4-d5qbv
[error]  Will try again in 5 seconds
[error]  Error restarting sync: Error selecting pod: Pod 'sc1-backend-api-65c45967f4-d5qbv' has critical status 'CrashLoopBackOff' 

To get more information about this pod's status, run: kubectl describe pod sc1-backend-api-65c45967f4-d5qbv
[error]  Will try again in 5 seconds
[error]  Error restarting sync: Error selecting pod: Pod 'sc1-backend-api-65c45967f4-d5qbv' has critical status 'CrashLoopBackOff' 

To get more information about this pod's status, run: kubectl describe pod sc1-backend-api-65c45967f4-d5qbv
[error]  Will try again in 5 seconds


```

Before 4.13 the logs showed me the issue, I was able to fix the code and the upload work fine and the container got restarted with a working version. 

Now I can fix the syntax error but the container keeps crashing since the sync seems no longer to work.

**What did you expect to happen instead?**  

`devspace dev` should be able to sync code inside a crash looping container as it worked in 4.12. 

I downgraded back to 4.12 and tested the same file change, and it worked:

```[sc1-backend-api] 
[sc1-backend-api] ############### Restart container ###############
[sc1-backend-api] 
[sc1-backend-api] # syscore.one/backend/api/pkg/api
[sc1-backend-api] api/pkg/api/project.go:226:1: syntax error: unexpected EOF, expecting }
[sc1-backend-api] 
[sc1-backend-api] Container exited with 2. Will restart in 7 seconds...
[sc1-backend-api] 
[sc1-backend-api] 
[sc1-backend-api] ############### Restart container ###############
[sc1-backend-api] 
[sc1-backend-api] 2020-06-30T17:51:23.85405259Z [DEBUG]: /app/shared/sentry/client.go:24 Sentry has been activated.
```

**How can we reproduce the bug?** (as minimally and precisely as possible)  

**Local Environment:**  
- DevSpace Version: 4.13.0
- Operating System: mac
- Deployment method: helm | kubectl apply

**Kubernetes Cluster:**  
- Cloud Provider: other
- Kubernetes Version: 
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-27T00:38:11Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

/kind bug

**Comment by FabianKramm** [comment]
@BirknerAlex thanks for reporting this issue! Good catch, this problem occurs because we renamed the sync helper binary to devspacehelper and the restart script now thinks the sync helper binary is missing and fails instead of just restarts the container.

**Comment by FabianKramm** [comment]
@BirknerAlex should be fixed with v4.13.1

**Comment by BirknerAlex** [comment]
@FabianKramm Thanks for the fast fix. Tested the new release, works fine! :heart:
